libretro-common/file/file_path.c

```diff
@@ -618,6 +629,91 @@ bool path_is_absolute(const char *path)
    return false;
 }
 
+bool is_posix_path(const char* path)
+{
+   if (strchr(path, posix_path_delimiter))
+      return true;
+
+   return false;
+}
+
+bool is_windows_path(const char* path)
+{
+   if (strchr(path, posix_path_delimiter))
+      return false;
+
+   return true;
+}
+
+/**
+ * path_resolve_to_local_file_system:
+ * @buf                    : output buffer
+ * @path                   : original relative path
+ * @base_content_directory : base content directory
+ *
+ * Resolves @path to local file system, switching '\' with '/' if needed (and viceversa)
+ * Appends @path to @base_content_directory
+ *
+ * Example[1]
+ *  @path = 'roms\zelda3.zip'
+ *  @base_content_directory = '/retroarch/'
+ *
+ *  Under linux/android returns '/retroarch/roms/zelda3.zip'
+ *
+ * Example[2]
+ *  @path = 'roms\zelda3.zip'
+ *  @base_content_directory = 'c:\games\retroarch\'
+ *
+ *  Under windows returns 'c:\games\retroarch\roms\zelda3.zip'
+ *
+ * Example[3]
+ *  @path = 'roms/zelda3.zip'
+ *  @base_content_directory = 'c:\games\retroarch\'
+ *
+ *  Under windows returns 'c:\games\retroarch\roms\zelda3.zip'
+ **/
+void path_resolve_to_local_file_system(char* buf, const char* path, const char* base_content_directory)
+{
```

Comment:
No, sorry - this needs to be rewritten. You absolutely cannot use the unsafe functions `strcpy()` and `strcat()`.

Please refer to the other routines in this file - they all take a `size_t len` argument, and only use safe functions.